server.js

```
@@ -0,0 +1,116 @@
+          "content-type": `text/${contentType}`
+        });
+        // write the actual file data
+        res.write(data.toString());
+        res.end();
+        return;
+      });
+    });
+  } // END OF GET
+
+  // START OF POST
+  if (req.method === "POST") {
+    let URL = req.url;
+    // body
+    let body = ""; // the data coming from Postman
+    req.on("data", chunk => {
+      body += chunk;
+
+      req.on("end", () => {
+        fs.writeFile(
+          `./public${URL}.html`,
+          elemTemplate(querystring.parse(body)), // read data as an object
+          err => {
+            if (err) {
+              return console.log(err);
+            }
+          }
+        );
+
+        fs.readdir(`./public`, (err, files) => {
+          // read the public folder directory
+          if (err) {
+            return console.log(err);
+          }
+
+          let filteredFiles = files.filter(
+            // take out anything that's not an element html file
+            element =>
+              !element.includes(".keep") &&
+              !element.includes("404.html") &&
+              !element.includes("index.html") &&
+              !element.includes("css")
+          );
+
+          let count = filteredFiles.length;
+          let listElement = ``;
+          let nameFormat = ``;
+
+          for (let i = 0; i < filteredFiles.length; i++) {
+            // formatting name and links for index.html
+            nameFormat =
+              filteredFiles[i].charAt(0).toUpperCase() +
+              filteredFiles[i].split(".")[0].slice(1);
+            listElement += `
+            <li>
+            <a href="/${filteredFiles[i]}">${nameFormat}</a>
+            </li>`;
+          }
+
+          let newIndex = indexTemplate(listElement, count);
+
+          fs.writeFile(`./public/index.html`, newIndex, err => {
+            // overwrite index file with updated index file
+            if (err) {
+              return console.log(err);
```

This should return an error to the client using either a 400/500 error